js/src/interface.js

```diff
@@ -274,6 +270,30 @@ let app = new Vue({
     }
   },
   methods: {
+    initDataSourceProvider(currentDataSourceId) {
+      const $vm = this;
+      let dataSourceData = {
+        dataSourceTitle: this.checkPackage ? 'Select the data source containing the user information' : 'Select data source',
```

Comment:
The data source query provider is used by various different components, e.g. validation manager and charts. This means the data source query provider should not have any idea of what it's being used for or what the data source title should be. I'm OK with `Select a data source` as a fallback, but the value should come from the component that's calling the data source query provider.

```suggestion
        dataSourceTitle: this.checkPackage
          ? params.dataSourceTitle // I'm assuming params is how you access the parameter passed to the provider
          : 'Select data source',
```

The same applies to the `default` parameter because the data source query provider should not have any confidence in calling a data source "Profile data" when it doesn't know what it'll be used for.

Reply:
@tonytlwu for the default parameter i should add the new parameter `dataSourceDefault` to the each component wich contains data-source-query provider, to pass it.

Reply:
@tonytlwu Now I put in a test value, I'm waiting for when we decide how to do it

Reply:
I hope you're not working on this provider in isolation. If you're working on this with one of the components, e.g. charts, then you should be able to remove the test value entirely.